iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -227,7 +230,7 @@ class string
     /// @endcode
     // TruncateToCapacity_t is a compile time variable to distinguish between constructors
     // NOLINTNEXTLINE(hicpp-named-parameter, readability-named-parameter)
-    string(TruncateToCapacity_t, const char* const other) noexcept;
+    string(TruncateToCapacity_t, const char* const other) noexcept(false);
```

Comment:
@saif-at-github Can you please try to replace the `std::cerr` in the implementation with our `LogError()`? This may resolves the issue and you can remove the `false` again. You can find an example for the usage of `LogError` in the `SharedMemoryObject`.

Reply:
@FerdinandSpitzschnueffler I had tried it, it doesn't resolve the issue.

Reply:
@saif-at-github I think it's because `ostream` can throw. I suggest to replace all occurrences of `std::cout` and `std::cerr` in the `cxx:string` with `LogWarn`/`LogError` and ignore the warning for now since @elBoberido's logger refactoring should not throw.

Reply:
@MatthiasKillat Since @elBoberido is on vacation, can you maybe confirm?

Reply:
@FerdinandSpitzschnueffler @saif-at-github I think keep the `noexcept` specifier as before for now. And yes, the streams can throw since they may allocate memory, so at least `bad_alloc` is possible.

I am not sure whether the tools deduce correctly that the exception specifier is incorrect. Probably because some function not marked as `noexcept` is called (which is too weak, as it only indicates that it cannot conclude that it does not throw).

TLDR: Keep the exception specifier for now, we need to take a closer look at the rule and maybe disable it.

Reply:
@elBoberido @saif-at-github If I am not mistaken the plan is to not change the exception specifier anywhere for now, even if it means there are still some Axivion findings.

We will decide how to deal with this separately as it affects the API on a semantic level.

Reply:
@FerdinandSpitzschnueffler @saif-at-github @MatthiasKillat good news everyone. With the merge of the logger this warning is gone on master :partying_face: